rootfs/etc/nginx/lua/util/timer.lua

```diff
@@ -0,0 +1,16 @@
+local new_timer = ngx.timer.at
+local _M = {}
+
+local function interval_thread(premature, interval, all_workers, func, ...)
+  if premature or ngx.worker.exiting() then return end
+  if all_workers or ngx.worker.id() == 0 then
+    func(...)
+    new_timer(interval, interval_thread, interval, all_workers, func, ...)
+  end
+end
+
+function _M.execute_at_interval(interval, all_workers, func, ...)
+  new_timer(0, interval_thread, interval, all_workers, func, ...)
+end
+
+return _M
```

Comment:
I'm not certain that's the case. This implementation leverages `ngx.timer.at` to execute at t=0,1*interval,2*interval... 

If you used `ngx.timer.every` you are only able to excute at t=1*interval,2*interval etc.

Reply:
https://github.com/openresty/lua-nginx-module#ngxtimerevery

Reply:
Looks like is because they wanted a zero-delay timer

Reply:
right but you can just do
```
f()
ngx.timer.every(interval, r)
```
instead of all those lines

also in the current code base I don't see reference to `execute_at_interval` at all

Reply:
Fair. Haha, but yes:

> also in the current code base I don't see reference to execute_at_interval at all

This is the real problem with this file yes? 😂Was being used by the other lua module I imported first.